File: app/src/main/java/com/rorpage/purtyweather/receivers/BootCompletedReceiver.java
package com.rorpage.purtyweather.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.rorpage.purtyweather.managers.ServiceManager;
import com.rorpage.purtyweather.util.WeatherUpdateScheduler;

public class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        ServiceManager.StartUpdateWeatherService(context);
        WeatherUpdateScheduler.scheduleJob(context);
    }
}

File: app/src/main/java/com/rorpage/purtyweather/receivers/BootCompletedReceiver.kt
package com.rorpage.purtyweather.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.google.accompanist.pager.ExperimentalPagerApi
import com.rorpage.purtyweather.util.WeatherUpdateScheduler

@ExperimentalPagerApi
class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        //TODO really we want to use the previous work request.
        WeatherUpdateScheduler.scheduleJob(context, false)
    }
}